engine-plugins/identity-ldap/src/main/java/org/camunda/bpm/identity/impl/ldap/LdapIdentityProviderSession.java

```diff
@@ -16,704 +16,879 @@
-    ensureContextInitialized();
+    protected void addFilter(String attributeName, String attributeValue, StringWriter writer) {
+        writer.write("(");
+        writer.write(attributeName);
+        writer.write("=");
+        writer.write(attributeValue);
+        writer.write(")");
+    }
+
+    protected LdapUserEntity transformUser(SearchResult result) throws NamingException {
+        final Attributes attributes = result.getAttributes();
+        LdapUserEntity user = new LdapUserEntity();
+        user.setDn(result.getNameInNamespace());
+        user.setId(getStringAttributeValue(ldapConfiguration.getUserIdAttribute(), attributes));
+        user.setFirstName(getStringAttributeValue(ldapConfiguration.getUserFirstnameAttribute(), attributes));
+        user.setLastName(getStringAttributeValue(ldapConfiguration.getUserLastnameAttribute(), attributes));
+        user.setEmail(getStringAttributeValue(ldapConfiguration.getUserEmailAttribute(), attributes));
+        return user;
+    }
+
+    protected GroupEntity transformGroup(SearchResult result) throws NamingException {
+        final Attributes attributes = result.getAttributes();
+        LdapGroupEntity group = new LdapGroupEntity();
+        group.setDn(result.getNameInNamespace());
+        group.setId(getStringAttributeValue(ldapConfiguration.getGroupIdAttribute(), attributes));
+        group.setName(getStringAttributeValue(ldapConfiguration.getGroupNameAttribute(), attributes));
+        group.setType(getStringAttributeValue(ldapConfiguration.getGroupTypeAttribute(), attributes));
+        return group;
+    }
+
+    /**
+     * Return the list of Controls requested in the query
+     *
+     * @param query query asks, contains the order by requested
+     * @return list of control to send to LDAP
+     */
+    private List<Control> getRequestControls(AbstractQuery<?, ?> query, StringBuilder resultLogger) {
+        try {
+            List<Control> controls = new ArrayList<>();
+
+            List<QueryOrderingProperty> orderBy = query.getOrderingProperties();
+
+            if (orderBy != null) {
+
+                for (QueryOrderingProperty orderingProperty : orderBy) {
+                    String propertyName = orderingProperty.getQueryProperty().getName();
+
+                    resultLogger.append(", OrderBy[");
+                    resultLogger.append(propertyName);
+                    resultLogger.append("-");
+                    resultLogger.append(orderingProperty.getDirection() == null ? "no_direction(desc)" : orderingProperty.getDirection().getName());
+                    resultLogger.append("]");
+
+                    SortKey sortKey = null;
+                    if (UserQueryProperty.USER_ID.getName().equals(propertyName)) {
+                        sortKey = new SortKey(ldapConfiguration.getUserIdAttribute(),
+                                Direction.ASCENDING.equals(orderingProperty.getDirection()),
```

Comment:
I understand, but it will not be very simple to do that. I think these two operations on the same PR save time. Are you really sure you want to go that path?

Reply:
Creating a new PR that provides your sorting fix is not too much work but helps us stay focused. This is also important for the downstream tasks of the development process (QA testing, documentation, etc.).

Reply:
Sounds good 👍 Looking forward for your second PR.

Reply:
Sounds good 👍 I'm looking forward to your second PR.